### PR TITLE
(SIMP-5799) Update puppetserver to use jruby9k  

### PR DESCRIPTION
**Pull request by jeannegreulich**
- updated the sysconfig file to use jruby 9k by default
      for PC1.  (PE already uses it.)
    - allow users to revert to jruby 1.7 by setting boolean,
      pupmod::master::jruby_alt, to false.
    - update tests
    - updated upper versions of concat and stdlib so the latest versions can
      be used.
    
    SIMP-5799 #close

### CODE REVIEW

File: templates/etc/sysconfig/puppetserver.epp [hunk 1]
```diff
@@ -25,6 +25,10 @@ JAVA_BIN="<%= $pupmod::master::sysconfig::java_bin %>"
 # Modify this if you'd like to change the memory allocation, enable JMX, etc
 JAVA_ARGS="-Xms<%= $java_start_memory %> -Xmx<%= $java_max_memory %> -Djava.io.tmpdir=<%= $pupmod::master::sysconfig::_java_temp_dir %> <%= $extra_java_args %>"
 
+<% if $pupmod::master::sysconfig::_use_jruby9k { -%>
```
Comment (op-ct) [inline templates/etc/sysconfig/puppetserver.epp]: _(This is a general observation, not a suggestion for this issue)_

An alternate approach to manage this setting, using [puppetlabs-inifile](https://github.com/simp/puppetlabs-inifile) (from [petems-hiera_vault](https://github.com/petems/petems-hiera_vault/blame/8b54fbe735d6bb9862c2f2eca6853e84b3518947/README.md#L31-L40)):

```puppet
ini_setting { "Change jruby to 9k":
  ensure            => present,
  setting           => 'JRUBY_JAR',
  path              => "/etc/sysconfig/puppetserver",
  key_val_separator => '=',
  section           => '',
  value             => '"/opt/puppetlabs/server/apps/puppetserver/jruby-9k.jar"',
  show_diff         => true,
  notify            => Service['puppetserver']
}
```

File: lib/facter/puppetserver_jruby.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    File.directory?('/opt/puppetlabs/server/apps/puppetserver') && File.readable?('/opt/puppetlabs/server/apps/puppetserver')
+  end
+
+  setcode do
+    jruby_hash = {
+      'dir' => '/opt/puppetlabs/server/apps/puppetserver',
+      'jarfiles' => []
+    }
+    jarfiles = Dir.glob("#{jruby_hash['dir']}/*.jar")
+    jruby_hash['jarfiles'] = jarfiles.map { |x| x.split('/').last}
```
Comment (lnemsick-simp) [inline lib/facter/puppetserver_jruby.rb]: File.basename(x) in lieu of x.split('/').last

File: CHANGELOG [hunk 3]
```diff
@@ -1,12 +1,16 @@
-* Wed Mar 06 2019 Liz Nemsick <lnemsick.simp@gmail.com> - 7.8.1-0
+* Thu Mar 07 2019 Jeanne Greulich <jeanne.greulich@onyxpoint.com> - 7.9.0-0
+- Update the default Jruby instance for puppetserver  opensource to jruby 9k.
+- Allow to select the old version if desired.
```
Comment (lnemsick-simp) [inline CHANGELOG]: Please note the new fact

**Comment by jeannegreulich** [comment]
https://gitlab.com/simp/pupmod-simp-pupmod/pipelines/51484047

Gitlab is still running.

File: CHANGELOG (hunk 3 above)
Comment (jeannegreulich) [reply CHANGELOG]: You changes look great.

**Pull request merged**